explorer/interpreter/action_stack.h

```diff
@@ -118,20 +120,30 @@ class ActionStack {
   // Suspends execution of the currently-executing continuation.
   auto Suspend() -> ErrorOr<Success>;
 
+  void Pop() { todo_.Pop(); }
+
  private:
   // Pop any ScopeActions from the top of the stack, propagating results as
   // needed, to restore the invariant that todo_.Top() is not a ScopeAction.
-  void PopScopes();
+  auto PopScopes() -> std::list<std::unique_ptr<Action>>;
 
   // Set `result` as the result of the Action most recently removed from the
   // stack.
   void SetResult(Nonnull<const Value*> result);
 
+  void DestructAllScopes();
+
+  void DestructScopes();
+
+  void PushCleanUpAction(std::unique_ptr<Action> act);
+
   // TODO: consider defining a non-nullable unique_ptr-like type to use here.
   Stack<std::unique_ptr<Action>> todo_;
   std::optional<Nonnull<const Value*>> result_;
   std::optional<RuntimeScope> globals_;
   Phase phase_;
+  std::optional<RuntimeScope> empty_locals_;
+  std::list<std::unique_ptr<Action>> destroy_actions_;
```

Comment:
Yes I can try... However this data structure is added because the Unwind Methodes call each other and with an Unwind Method I can only put the CleanUps on the stack after SetResult has been executed....
That doesn't mean you can't do it better....